agent/xds/clusters.go

```diff
@@ -47,8 +47,14 @@ func (s *Server) clustersFromSnapshotConnectProxy(cfgSnap *proxycfg.ConfigSnapsh
 	// TODO(rb): this sizing is a low bound.
 	clusters := make([]proto.Message, 0, len(cfgSnap.Proxy.Upstreams)+1)
 
+	proxyCfg, err := ParseProxyConfig(cfgSnap.Proxy.Config)
+	if err != nil {
+		s.Logger.Warn("failed to parse Connect.Proxy.Config", "error", err)
```

Comment:
This may need to return an error, as a I discovered the previous call claimed to not return the error, but was at the end of `makeAppCluster`